src/go/rpk/pkg/cli/cmd/plugin/plugin.go

```diff
@@ -0,0 +1,297 @@
+`,
+
+		Args: cobra.ExactArgs(0),
+		Run: func(*cobra.Command, []string) {
+			installed := plugin.ListPlugins(fs, plugin.UserPaths())
+
+			if local {
+				installed.Sort()
+
+				tw := out.NewTable("NAME", "PATH", "SHADOWS")
+				defer tw.Flush()
+
+				for _, p := range installed {
+					var shadowed string
+					if len(p.ShadowedPaths) > 0 {
+						shadowed = p.ShadowedPaths[0]
+					}
+					tw.Print(p.Name(), p.Path, shadowed)
+
+					if len(p.ShadowedPaths) < 1 {
+						continue
+					}
+					for _, shadowed = range p.ShadowedPaths[1:] {
+						tw.Print("", "", shadowed)
+					}
+				}
+
+				return
+			}
+
+			m, err := getManifest()
+			out.MaybeDieErr(err)
+
+			tw := out.NewTable("NAME", "DESCRIPTION", "MESSAGE")
+			defer tw.Flush()
+			for _, entry := range m.Plugins {
+				name := entry.Name
+				_, entrySha, _ := entry.PathShaForUser()
+
+				var message string
+
+				p, exists := installed.Find(name)
+				if exists {
+					name = "*" + name
+
+					sha, err := plugin.Sha256Path(fs, p.Path)
+					if err != nil {
+						message = fmt.Sprintf("unable to calculate local binary sha256: %v", err)
+					} else if sha != entrySha {
+						message = "local binary sha256 differs from manifest sha256"
+					}
+				}
+
+				tw.Print(name, entry.Description, message)
+			}
+		},
+	}
+
+	cmd.Flags().BoolVarP(&local, "local", "l", false, "list locally installed plugins and shadowed plugins")
+
+	return cmd
+}
+
+func newInstallCommand(fs afero.Fs) *cobra.Command {
+	var dir string
+	var update bool
+	cmd := &cobra.Command{
+		Use:     "install [PLUGIN]",
+		Aliases: []string{"download"},
+		Short:   "Install an rpk plugin.",
+		Long: `Install an rpk plugin.
+
+An rpk plugin must be saved in a directory that is in your $PATH. By default,
+this command installs plugins to the first directory in your $PATH. This can
+be overridden by specifying the --bin-dir flag.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(_ *cobra.Command, args []string) {
+			name := args[0]
+
+			fmt.Printf("Searching plugin manifest for %q...\n", name)
+			m, err := getManifest()
+			out.MaybeDieErr(err)
+
+			p, err := m.FindEntry(name)
+			out.MaybeDieErr(err)
+
+			_, remoteSha, err := p.PathShaForUser()
+			out.MaybeDieErr(err)
+
+			var found bool
+			installed := plugin.ListPlugins(fs, plugin.UserPaths())
+			for _, p := range installed {
+				if name == p.Name() {
+					sha, err := plugin.Sha256Path(fs, p.Path)
+					out.MaybeDieErr(err)
+
+					if sha != remoteSha {
+						msg := fmt.Sprintf(`Plugin %q is already installed, but is different from the remote plugin!
+
+ Local sha256: %s
+Remote sha256: %s
+
+`, name, sha, remoteSha)
+						if !update {
+							out.Exit(msg + "--update was not requested, exiting!")
+						}
+						fmt.Println(msg + "Downloading and validating updated plugin...")
+						found = true
+
+					} else {
+						out.Exit("Plugin %q is already installed and up to date!", name)
```

Comment:
Having small else clauses makes it a little harder to read code. Can you invert the condition and exit early, so the reader can forget about the "easier" path and remove the extra nesting?